src/config.cpp

```diff
@@ -421,6 +430,18 @@ void parse_program_options(int argc, const char** argv)
         }
     }
 
+    if (arguments.given("as-pre-sudo-user") && std::getenv("SUDO_UID") == NULL)
```

Comment:
This is not the behavior I would expect.
While `lo2s -a -A` works fine, `lo2s -u -U $USER` complains about `-u was specified but no sudo was detected.`

Or in other words, `-U` should imply `-u`.

Reply:
Also, in C++, we don't use `NULL`. Use `nullptr` instead.